api/samlacs.go

```diff
@@ -148,7 +154,24 @@ func (a *API) SAMLACS(w http.ResponseWriter, r *http.Request) error {
 		}
 	}
 
-	// TODO: fetch new metadata if possible when validUntil < time.Now() or cacheDuration
+	var samlMetadataModified bool
+	samlMetadataModified = false
+	if IsMetadataStale(idpMetadata, ssoProvider.SAMLProvider) {
+		if *ssoProvider.SAMLProvider.MetadataURL != "" {
```

Comment:
I guess these two conditions can be one. `IsMetadataStale` should not even be called if there's no MetadataURL on the provider itself.